services/network.proto

```diff
@@ -0,0 +1,13 @@
+message ServerIdentity{
+    required bytes public = 1;
+    required bytes id = 2;
+    required string address = 3;
+    required string description = 4;
```

Comment:
Does it "simply" work if you put it as optional then ? I don't see any point in having something required when it's actually optional. As a general guideline , we should be careful to put the least possible fields